From 6c5581808deaa1cba47eedf6479571c221443c6e Mon Sep 17 00:00:00 2001
From: Michael Auchter <michael.auchter@ni.com>
Date: Tue, 28 Jan 2020 14:50:16 -0600
Subject: [PATCH 14/37] board: ni: update x410 configs

- Update default config to use the same environment storage as Mender by
  default.
- Add sdboot option and simplify the Mender integration

Signed-off-by: Michael Auchter <michael.auchter@ni.com>
---
 configs/ni_x410_rev2_defconfig |  6 +++++-
 configs/ni_x410_rev3_defconfig |  6 +++---
 include/configs/ni-x410.h      | 18 ++++++++++++++----
 3 files changed, 22 insertions(+), 8 deletions(-)

diff --git a/configs/ni_x410_rev2_defconfig b/configs/ni_x410_rev2_defconfig
index 317d0204175..9d53e2fa305 100644
--- a/configs/ni_x410_rev2_defconfig
+++ b/configs/ni_x410_rev2_defconfig
@@ -4,6 +4,8 @@ CONFIG_SYS_CONFIG_NAME="ni-x410"
 CONFIG_ARCH_ZYNQMP=y
 CONFIG_SYS_TEXT_BASE=0x8000000
 CONFIG_SYS_MALLOC_F_LEN=0x8000
+CONFIG_ENV_SIZE=0x20000
+CONFIG_ENV_OFFSET=0x800000
 CONFIG_SPL=y
 CONFIG_DEBUG_UART_BASE=0xff000000
 CONFIG_DEBUG_UART_CLOCK=100000000
@@ -51,10 +53,12 @@ CONFIG_CMD_TIMER=y
 CONFIG_CMD_EXT4_WRITE=y
 CONFIG_SPL_OF_CONTROL=y
 CONFIG_DEFAULT_DEVICE_TREE="ni_x410-rev2"
-CONFIG_ENV_IS_IN_FAT=y
+CONFIG_ENV_IS_IN_MMC=y
 CONFIG_NET_RANDOM_ETHADDR=y
 CONFIG_SPL_DM=y
 CONFIG_SPL_DM_SEQ_ALIAS=y
+CONFIG_BOOTCOUNT_LIMIT=y
+CONFIG_BOOTCOUNT_ENV=y
 CONFIG_CLK_ZYNQMP=y
 CONFIG_DFU_RAM=y
 CONFIG_FPGA_XILINX=y
diff --git a/configs/ni_x410_rev3_defconfig b/configs/ni_x410_rev3_defconfig
index e78caff4655..ee7f873ad45 100644
--- a/configs/ni_x410_rev3_defconfig
+++ b/configs/ni_x410_rev3_defconfig
@@ -4,6 +4,8 @@ CONFIG_SYS_CONFIG_NAME="ni-x410"
 CONFIG_ARCH_ZYNQMP=y
 CONFIG_SYS_TEXT_BASE=0x8000000
 CONFIG_SYS_MALLOC_F_LEN=0x8000
+CONFIG_ENV_SIZE=0x20000
+CONFIG_ENV_OFFSET=0x800000
 CONFIG_SPL=y
 CONFIG_DEBUG_UART_BASE=0xff000000
 CONFIG_DEBUG_UART_CLOCK=100000000
@@ -53,12 +55,10 @@ CONFIG_CMD_TIMER=y
 CONFIG_CMD_EXT4_WRITE=y
 CONFIG_SPL_OF_CONTROL=y
 CONFIG_DEFAULT_DEVICE_TREE="ni_x410-rev3"
-CONFIG_ENV_IS_IN_FAT=y
-CONFIG_ENV_FAT_DEVICE_AND_PART="1:auto"
+CONFIG_ENV_IS_IN_MMC=y
 CONFIG_NET_RANDOM_ETHADDR=y
 CONFIG_SPL_DM=y
 CONFIG_SPL_DM_SEQ_ALIAS=y
-# CONFIG_BLOCK_CACHE is not set
 CONFIG_BOOTCOUNT_LIMIT=y
 CONFIG_BOOTCOUNT_ENV=y
 CONFIG_CLK_ZYNQMP=y
diff --git a/include/configs/ni-x410.h b/include/configs/ni-x410.h
index 20455a4ea55..d74342809cb 100644
--- a/include/configs/ni-x410.h
+++ b/include/configs/ni-x410.h
@@ -16,7 +16,8 @@
         "ec_image=lib/firmware/ni/ec-titanium-rev2.RW.bin\0" \
         "expand_ec_image=setenv ec_image lib/firmware/ni/ec-titanium-rev${ec_compat}.RW.bin\0" \
         "load_addr=0x2000000\0" \
-        "bootargs=uio_pdrv_genirq.of_id=usrp-uio at24.io_limit=64 rootwait clk_ignore_unused\0" \
+        "fit_image=boot/fitImage\0" \
+        "bootargs=uio_pdrv_genirq.of_id=usrp-uio at24.io_limit=64\0" \
         "emmcboot=" \
           "setenv disk_name eMMC && " \
           "setenv uboot_root \"mmc 0:2\" && " \
@@ -32,14 +33,14 @@
           "setenv bootargs root=${kernel_root} ${bootargs} && " \
           "load ${uboot_root} $load_addr ${fit_image} && " \
           "bootm ${load_addr}#conf@${variant}.dtb${overlays}\0" \
-        "swsync_emmc=if test ${ec_disable_swsync} = 0; then " \
-          "echo Copying MCU FW from eMMC to RAM... && " \
+        "swsync_disk=if test ${ec_disable_swsync} = 0; then " \
+          "echo Copying MCU FW from ${disk_name} to RAM... && " \
           "run expand_ec_image; " \
           "load ${uboot_root} ${load_addr} ${ec_image} && " \
           "crosec swsync ${load_addr} ${filesize}; fi;\0" \
         "netargs=printenv serverip && printenv tftproot && printenv nfsroot" \
           " && setenv bootargs root=/dev/nfs nfsroot=${serverip}:${nfsroot},v3,tcp" \
-          " ip=dhcp uio_pdrv_genirq.of_id=usrp-uio at24.io_limit=64\0" \
+          " ip=dhcp ${bootargs}\0" \
         "netboot=run netargs && run swsync_net && " \
           "echo Loading FIT to RAM via DHCP/TFTP && " \
           "dhcp ${load_addr} ${tftproot}/${fit_image} && "\
@@ -56,4 +57,13 @@
 
 #define CONFIG_TFTP_PORT
 
+#define CONFIG_ENV_OFFSET_REDUND 0x1000000
+#define CONFIG_SYS_MMC_ENV_PART 0
+
+#ifdef CONFIG_SPL_ZYNQMP_TWO_SDHCI
+#define CONFIG_SYS_MMC_ENV_DEV 1
+#else
+#define CONFIG_SYS_MMC_ENV_DEV 0
+#endif
+
 #endif /* __CONFIG_ETTUS_X410_H */
-- 
2.43.0

